lib/matplotlib/markers.py

```diff
@@ -363,11 +391,101 @@ def get_alt_transform(self):
         Return the transform to be applied to the `.Path` from
         `MarkerStyle.get_alt_path()`.
         """
-        return self._alt_transform.frozen()
+        if self._user_transform is not None:
+            return (self._alt_transform + self._user_transform).frozen()
+        else:
+            return self._alt_transform.frozen()
 
     def get_snap_threshold(self):
         return self._snap_threshold
 
+    def get_user_transform(self):
+        """Return user supplied part of marker transform."""
+        if self._user_transform is not None:
+            return self._user_transform.frozen()
+
+    def transformed(self, transform: Affine2D):
+        """
+        Return new marker with combined transformation.
+
+        Parameters
+        ----------
+        transform : Affine2D, default: None
+            Transform will be combined with current user supplied transform.
+        """
+        new_marker = MarkerStyle(self)
+        if new_marker._user_transform is not None:
+            new_marker._user_transform += transform
+        else:
+            new_marker._user_transform = transform
+        return new_marker
+
+    def rotated(self, deg=None, rad=None):
+        """
+        Return new marker rotated by specified angle.
+
+        Parameters
+        ----------
+        deg : float, default: None
+            Use this parameter to specify rotation angle in degrees.
+
+        rad : float, default: None
+            Use this parameter to specify rotation angle in radians.
+
+        .. note:: You must specify exactly one of deg or rad.
+        """
+        if not ((deg is None) ^ (rad is None)):
+            raise ValueError("Exactly one of deg or rad shall be used.")
+
+        new_marker = MarkerStyle(self)
+        if new_marker._user_transform is None:
+            new_marker._user_transform = Affine2D()
+
+        if deg is not None:
+            new_marker._user_transform.rotate_deg(deg)
+        if rad is not None:
+            new_marker._user_transform.rotate(rad)
+
+        return new_marker
+
+    def scaled(self, sx, sy=None):
+        """
+        Return new marker scaled by specified scale factors.
+
+        If *sy* is None, the same scale is applied in both the *x*- and
+        *y*-directions.
+
+        Parameters
+        ----------
+        sx : float
+            *X*-direction scaling factor.
+        sy : float, default: None
+            *Y*-direction scaling factor.
+        """
+        if sy is None:
+            sy = sx
```

Comment:
It looks like the scaling code needs a test adding to check it's working fine.

Reply:
Test added, I am now aware of the coverage results, thanks.